libbpf-rs/src/map.rs

```diff
@@ -602,6 +602,11 @@ impl Map {
             Link::new(ptr)
         })
     }
+
+    /// Retrieve the underlying [`libbpf_sys::bpf_map`].
+    pub fn as_libbpf_bpf_map_ptr(&self) -> Option<NonNull<libbpf_sys::bpf_map>> {
```

Comment:
curious why this one returns Option<> while as_libbpf_bpf_object_ptr() doesn't?

Reply:
See,
https://github.com/heyrutvik/libbpf-rs/blob/f9b3d9d01b467c1a561a49dc2c40d7a235926c0f/libbpf-rs/src/map.rs#L161-L162 

The `ptr` could be null if the map is created from userspace. I kept the return type `Option<_>` since I was not sure how to handle the call if `ptr` is not updated with correct value.

Reply:
@anakryiko Let me know if you have any suggestions about it.